### PR TITLE
React settings

### PR DESCRIPTION
**Pull request by Manuel-777**
Pushing this PR before @lusbenjamin hehe

I initially added a `Checkbox` component and the Behaviour panel, but some more base components are missing before converting everything, like `Input` and `Slider`.

### CODE REVIEW

**Comment by Manuel-777** [comment]
This is now ready for ewview, just some things that need to be fixed at first glance:

- In Visual settings, the card tile preview is always Arena despite the selected mode.
- Clicking on any setting makes the whole react tree in settings to reload, causing the view to move to top scroll and destroying the state and all. This is caused by the IPC message mounting the settings component again. It should be easy to fix once the whole renderer is React, now the only way (I think) is making the settings update ipc message to not sync back to renderer.

File: src/shared/ReactSelect.tsx [hunk 1]
```diff
@@ -69,10 +70,12 @@ export interface WrappedReactSelectProps extends ReactSelectProps {
 
 // This is essentially what createSelect does, but reacty.
 // This should go away once createSelect goes away and is replaced by just ReactSelect.
-export function WrappedReactSelect(props: WrappedReactSelectProps) {
+export function WrappedReactSelect(
+  props: WrappedReactSelectProps
+): JSX.Element {
   const { className, ...other } = props;
   return (
-    <div className={"select_container " + className}>
```
[the remark this answers is not in the document]
Comment (lusbenjamin) [reply src/shared/ReactSelect.tsx]: Explore page regression: 
image

**Comment by lusbenjamin** [comment]
Took this for a spin and it looks pretty great so far! I fixed all of the problems I found, but there may still be a few more lurking around. We still need to solve the visual regression of `ReactSelect` on the legacy pages.

**Comment by lusbenjamin** [comment]
> "Clicking on any setting makes the whole react tree in settings to reload, causing the view to move to top scroll and destroying the state and all. This is caused by the IPC message mounting the settings component again. It should be easy to fix once the whole renderer is React, now the only way (I think) is making the settings update ipc message to not sync back to renderer."

This is now solved for all settings pages except the Overlay settings. I did not confirm this, but I _think_ we have to force a full refresh in those cases to ensure that the updates are pushed to the other overlay renderer process. In all other cases where we could potentially skip the refresh, we now do (updating app settings, updating main renderer settings).

File: src/shared/ReactSelect.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lusbenjamin) [reply src/shared/ReactSelect.tsx]: solved by reverting to legacy-compatible props

**Comment by lusbenjamin** [comment]
@Manuel-777 you probably have incorrect settings polluting your `settings.json` (app DB file), since this PR used to be persisting several user-level settings to the app DB incorrectly.

**Comment by Manuel-777** [comment]
> @Manuel-777 you probably have incorrect settings polluting your `remember.json` (app DB file), since this PR used to be persisting several user-level settings to the app DB incorrectly.

You're correct! Silly copy pasta bug, good catch! 🏆

**Pull request merged**